addon/components/tags-widget/component.js

```diff
@@ -49,22 +49,28 @@ export default Ember.Component.extend({
                     this.send('removeATag', tag);
                 }
             });
-            // Populate the widget with the default tags passed in
-            let tags = this.get('tags');
-            this.$().importTags(tags.join(', '));
         });
     }),
 
+    didRender() {
+        // Rerender the list of tags whenever the node model changes. Useful if node.tags is not defined when page loads.
+        // Provide a default value in case tags weren't defined when component first rendered
+        this.$().importTags('');
+        let tags = this.get('tags') || [];
+        this.$().importTags(tags.join(', '));
+    },
+
     actions: {
         addATag(tag) {
+            // Calls a curried closure action which was provided the model
             this.attrs.addATag(tag);
         },
         removeATag(tag) {
             // Don't try to delete a blank tag (would result in a server error)
             if (!tag) {
                 return false;
             }
-            this.sendAction('removeATag', tag);
```

Comment:
We're calling it this way to take advantage of a quirk of closure actions. Calling `.send` or `.sendAction` is a fire-and-forget process, but there are times we want our components to respond to the result of an action. (eg to clean up internal state for the component if a promise/save fails) 

In the future we may choose to respond to the result of a failed promise in deciding how to update a context-specific save failure message field in the component, for example.

https://alisdair.mcdiarmid.org/ember-closure-actions-have-return-values/